test/demo/contextChannels.test.ts

```diff
@@ -246,6 +246,32 @@ describe('When joining a channel', () => {
         expect(payload[0]).toHaveProperty('previousChannel.id', 'blue');
         expect(payload[0]).toHaveProperty('identity', channelChangingWindow);
     });
+
+    it('if everything is unsubscribed, and something rejoins, there is no data left over', async ()=>{
+        const [greenWindow1, channelChangingWindow] = await setupWindows('green', 'yellow');
```

Comment:
From the test description, it seems one channel would suffice for this - is there a reason for using two?

If there's some subtle edge case being covered here, would be good to make that explicit - or even a separate test.

Reply:
Ah OK. So this test could probably do with some docs. What I am testing is this:

1 - if I join a channel where there is data, I get the listener called 
2 - if I join a channel where there is no data, I get no listener callback.

Which are obviously just two sides of the same coin. 1 is quite hard to test because of the way the setupWindows function joins the channel - I miss the first callback, because it has already happened before I register. So I set up in some dummy channel (yellow) and then join green, and get my callback. Then I clear the green channel by switching everyone away, and when I rejoin I can test 2.

So, Phil, is your question, should I just use yellow rather than global so I have two channels rather than getting global involved?

Reply:
I think I have done what you asked